apps/src/applab/designElements/radioButton.jsx

```diff
@@ -111,17 +114,76 @@ var RadioButtonEvents = React.createClass({
   }
 });
 
+/**
+ * Gets the initial group id for a new radio button.
+ * To figure out the initial group id, we:
+ * 1) Try to find the most recently created radio button
+ *    on the current screen.
+ * 2) If it exists, use that group id. If not, generate an unused group id.
+ * @returns {string} The default group id for the new radio button
+ */
+function getInitialGroupId() {
+  // Get the most recently added button on the current screen
+  var lastRadioButton = getLastRadioButtonOnCurrentScreen();
+
+  if (lastRadioButton && lastRadioButton.getAttribute('name') && 
+      lastRadioButton.getAttribute('name').trim() !== '') {
+
+    // We have an existing radio button, use that group id
+    return lastRadioButton.getAttribute('name');
+  }
+
+  // Otherwise, generate an unused one
+  return getUnusedGroupId();
+}
+
+/**
+ * Gets the most recently added radio button on current screen.
+ * @returns {HTMLElement} The radio button element. Returns null if none exists.
+ */
+function getLastRadioButtonOnCurrentScreen() {
+  // Get the current visible screen element
+  var currentScreen = $('#designModeViz .screen:visible').first();
+
+  // Find all the radio button elements on the current screen, if any
+  var radioButtons = currentScreen.find('input[type=radio]');
+
+  if (radioButtons.length > 0) {
+    // Return the last radio button
+    return radioButtons[radioButtons.length - 1];
+  }
+
+  return null;
+}
+
+/**
+ * Generates a group id that is not used by any other existing radio buttons.
+ * @returns {string} An group id that isn't used by other radio buttons
+ */
+function getUnusedGroupId() {
+  var i = 1;
+  while ($('input[name=' + GROUP_ID_PREFIX + i + ']').length > 0) {
+    i++;
+  }
+
+  return GROUP_ID_PREFIX + i;
+}
```

Comment:
Yeah, I was looking at `getUnusedElementId` but realize I'll have to add a fork and won't be actually be able to reuse most of that method. It also felt nicer to contain this logic within the radio button component, since group ids is a thing unique to that control.